tests/framework/report.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+'''
+This file implements functionality for generating reports.
+
+A report typically consists of what tests have been executed, the type of test,
+what functionality it validates, what is the criteria that we use to validate
+and an optional link to a GitHub issue.
+
+To pick up all the items listed above, this module defines the `Report` class
+which is responsible for gathering data on what tests have been executed,
+extracting test information and generating the report.
+
+While some items such as test name can be extracted directly from code, others
+such as test type or issue link are extracted from the function docstring.
+A test docstring is defined like below:
+    def test_net_device():
+        """
+        Test description
+        @type: regression
+        @issue: https://github.com/firecracker-microvm/firecracker/issues/1
+        """
```

Comment:
I think it would be useful to make this example a bit more complex, maybe containing all the tokens available.